securedrop/journalist.py

```diff
@@ -120,38 +120,53 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def validate_user(username, password, token, error_message=None):
+    """
+    Validates the user by calling the login and handling exceptions
+    :param username: Username
+    :param password: Password
+    :param token: Two-factor authentication token
+    :param error_message: Localized error message string to use on failure
+    :return: Journalist user object if successful, None otherwise.
+    """
+    try:
+        return Journalist.login(username, password, token)
+    except Exception as e:
```

Comment:
I think we can this a bit more specific and only catch the crypto / no result cases. The normal 500 error handler will catch everything else.

This could be `except (InvalidUsernameException, BadTokenException, WrongPasswordException)`.

Sorry I didn't catch this earlier before we did 10,000 CircleCI builds. :/

Reply:
The suggestion from @heartsucker sounds good to me.

Reply:
I agree with you both @heartsucker @kushaldas that we should only catch what we handle here and let 500s occur for any other exceptions. That said, since in this PR the author moved the existing validation code into its own function for reuse, I'm cool with making a followup issue to address improving the exception handling in this function.

Reply:
Ok, marking this as approved. Whoever merges this should make the follow up issue.